allennlp/tests/models/bert_srl_test.py

```diff
@@ -0,0 +1,56 @@
+# pylint: disable=no-self-use,invalid-name
+import numpy
+from _pytest.monkeypatch import MonkeyPatch
+from pytorch_pretrained_bert.modeling import BertConfig, BertModel
+from pytorch_pretrained_bert.tokenization import BertTokenizer
+
+from allennlp.common.testing import ModelTestCase
+from allennlp.nn.util import get_lengths_from_binary_sequence_mask
+
+
+class BertSrlTest(ModelTestCase):
+    def setUp(self):
+
+        self.monkeypatch = MonkeyPatch()
+        # monkeypatch the PretrainedBertModel to return the tiny test fixture model
+        config_path = self.FIXTURES_ROOT / 'bert' / 'config.json'
+        vocab_path = self.FIXTURES_ROOT / 'bert' / 'vocab.txt'
+        config = BertConfig(str(config_path))
+        self.monkeypatch.setattr(BertModel, 'from_pretrained', lambda _: BertModel(config))
+        self.monkeypatch.setattr(BertTokenizer, 'from_pretrained', lambda _: BertTokenizer(vocab_path))
+
+        super(BertSrlTest, self).setUp()
+        self.set_up_model(self.FIXTURES_ROOT / 'bert_srl' / 'experiment.jsonnet',
+                          self.FIXTURES_ROOT / 'conll_2012')
+
+    def tearDown(self):
+        self.monkeypatch.undo()
+        self.monkeypatch.undo()
```

Comment:
why are there 2 undos here?

actually, the monkeypatch documentation claims that manually calling undo is not usually necessary, why do we need to do that here?

Reply:
I thought it was because 2 things are monkeypatched and they keep a stack of the patches. I also thought that it was unnecessary, but without it several tests failed because the monkeypatch was still there, and with this it doesn't 🤷‍♂